nucypher/policy/conditions/evm.py

```diff
@@ -168,41 +169,64 @@ def _next_endpoint(
             # each endpoint here to see if it's alive and only yield it if it is.
             yield provider
 
+    def _execute_with_provider(
+        self,
+        provider: HTTPProvider,
+        resolved_parameters: List[Any],
+        check_chain: bool = True,
+    ) -> Any:
+        w3 = self._configure_w3(provider)
+        if check_chain:
+            self._check_chain_id(w3)
+        return self._execute(w3, resolved_parameters)
```

Comment:
The checking of the chain should always be done. It ensures that the provider's chain id and condition chain id always match up - this is irrespective of whether the customized rpc_endpoint is provided or not.

```suggestion
    def _execute_with_provider(
        self,
        provider: HTTPProvider,
        resolved_parameters: List[Any],
    ) -> Any:
        w3 = self._configure_w3(provider)
        self._check_chain_id(w3)
        return self._execute(w3, resolved_parameters)
```

^ callers will need to be updated.

Reply:
fixed :)